doc/vrrp/VRRP_Adaptation_HLD.md

```diff
@@ -0,0 +1,712 @@
+    - Add/del VIP to Macvlan device;
+    - Upadate VRRP instance configuration to the APPL DB.
+
+vrrpsyncd: 
+  - Complete the following tasks:
+    - Listen for kernel Macvlan device and other interfaces protocol status change events
+    - Update the kernel Macvlan device status to the APPL DB.
+    - Match VRRP instance tracking interface and recalculate priority.
+    - Update priority change to vrrpd by using FRR CLI commands.
+
+##### SWSS container
+
+vrrporch: 
+  - Subscribes to APPL_DB tables, responsible for updating the ASIC DB. Creating/deleting Virtual RIF object via SAI API.
+
+![VRRP Vrrporch Uml](images/VRRP_Vrrporch_Uml.png)
+
+#### CoPP Configurations
+
+CoPP will be extended as follows for trapping VRRPs:
+```
+  "trap.group.vrrp": {
+    "cir":"300",
+    "cbs":"300",
+    "meter_type":"packets",
+    "mode":"sr_tcm",
+    "red_action":"drop",
+    "trap_action":"trap",
+    "trap_priority":"7",
+    "queue": "7",
+    "yellow_action":"drop"
+  },
+  "trap.group.vrrpv6": {
+    "cir":"300",
+    "cbs":"300",
+    "meter_type":"packets",
+    "mode":"sr_tcm",
+    "red_action":"drop",
+    "trap_action":"trap",
+    "trap_priority":"7",
+    "queue": "7",
+    "yellow_action":"drop"
+  }
+  ...
+  "vrrp": {
+    "trap_ids": "vrrp",
+    "trap_group": "trap.group.vrrp",
+    "always_enabled": "true"
+  },
+  "vrrpv6": {
+    "trap_ids": "vrrpv6",
+    "trap_group": "trap.group.vrrpv6",
+    "always_enabled": "true"
+  }
+```
+#### DB Changes
+
+##### CONFIG_DB changes
+
+VRRP
+
+Producer: config manager
+
+Consumer: vrrpmgrd and vrrpcfgd/frrcfgd
+
+Description: New table that stores VRRP configuration for per interface + VRID.
+```
+;New table
+;holds the VRRP configuration per interface and VRID
+key             = VRRP:interface_name:vrid
+                                                ; Interface name string. Vlan, Ethernet or PortChannel
+                                                ; vrid is an integer
+; field = value
+vip             = ip_address                    ; Virtual IP address. This is a list of IP addresses
+priority        = 3DIGIT                        ; Priority of VRRP instance
+adv_interval    = 5DIGIT                        ; Advertisement interval for VRRP. Default = 1000ms
+state           = vrrp_state                    ; String denoting the state of VRRP instance
+version         = vrrp_version                  ; VRRP version. Value is 2 or 3. Default is 3
+preempt         = "enabled"/"disabled"          ; VRRP pre-emption is enabled? Default is enabled
+track_interface = interface_Name|weight_value   ;This is repeated for the configured tracking interfaces
+```
+Example:-
+```
+admin@sonic:~$ redis-cli -n 4 keys VRRP*
+
+"VRRP|Ethernet8|8"
+"VRRP|Ethernet1|1"
+"VRRP|Vlan10|10"
+"VRRP|Ethernet2|2"
+"VRRP|Vlan4|4"
+
+admin@sonic:~$ redis-cli -n 4 HGETALL " VRRP|Vlan10|10"
+
+"vrid"
+"10"
+"vip"
+"4.1.1.100/24,4.1.1.200/24,2000::50/64"
+"priority"
+"100"
+"adv_interval"
+"1000"
+"state"
+""
```

Comment:
Can the state be empty?

Reply:
No, I will correct it. Default is up. thanks!